thematic_areas/views.py

```diff
@@ -9,7 +9,96 @@
 
 from core.libs import chkcsv
 
-from .models import ThematicAreaFile, ThematicArea
+from .models import ThematicAreaFile, ThematicArea, GenericThematicAreaFile, GenericThematicArea
+
+
+def generic_validate(request):
+    """
+    This view function validade a csv file based on a pre definition os the fmt
+    file.
+
+    The check_csv_file function check that all of the required columns and data
+    are present in the CSV file, and that the data conform to the appropriate
+    type and other specifications, when it is not valid return a list with the
+    errors.
+    """
+    errorlist = []
+    file_id = request.GET.get("file_id", None)
+
+    if file_id:
+        file_upload = get_object_or_404(GenericThematicAreaFile, pk=file_id)
+
+    if request.method == 'GET':
+        try:
+            upload_path = file_upload.attachment.file.path
+            cols = chkcsv.read_format_specs(
+                os.path.dirname(os.path.abspath(__file__)) + "/chkcsvfmt.fmt", True, False)
+            errorlist = chkcsv.check_csv_file(upload_path, cols, True, True, True, False)
+            if errorlist:
+                raise Exception(_("Validation error"))
+            else:
+                file_upload.is_valid = True
+                fp = open(upload_path)
+                file_upload.line_count = len(fp.readlines())
+                file_upload.save()
+        except Exception as ex:
+            messages.error(request, _("Validation error: %s") % errorlist)
+        else:
+            messages.success(request, _("File successfully validated!"))
+
+    return redirect(request.META.get('HTTP_REFERER'))
+
+
+def generic_import_file(request):
+    """
+    This view function import the data from a CSV file.
+
+    Something like this:
+
+        Title,Institution,Link,Description,date
+        Politica de acesso aberto,Instituição X,http://www.ac.com.br,Diretório internacional de política de acesso aberto
+
+    TODO: This function must be a task.
+    """
+    file_id = request.GET.get("file_id", None)
+
+    if file_id:
+        file_upload = get_object_or_404(GenericThematicAreaFile, pk=file_id)
+
+    file_path = file_upload.attachment.file.path
+
+    try:
+        with open(file_path, 'r') as csvfile:
+            data = csv.DictReader(csvfile, delimiter=";")
+
+            for line, row in enumerate(data):
+                the_area = GenericThematicArea()
+                the_area.text = row.get('text')
+                the_area.lang = row.get('lang')
+                the_area.origin = row.get('origin')
+                the_area.level = row.get('level')
+                the_area.creator = request.user
+                the_area.save()
+
+    except Exception as ex:
+        messages.error(request, _("Import error: %s, Line: %s") % (ex, str(line + 2)))
```

Comment:
@Rossi-Luciano coloque isso dentro do `for`, pois se ocorrer erro na linha 71, nem existirá a variável `line`.

Reply:
O código foi corrigido.